### PR TITLE
More config cleanup

### PR DESCRIPTION
**Pull request by richard-rogers**
* All the `init()` functions take an optional `LangKitConfig` argument
* All magic strings moved to `LangKitConfig`
* Topic UDF directly callable
* Other minor cleanup

### CODE REVIEW

File: poetry.lock
Comment (jamie256) [inline poetry.lock:1]: revert poetry.lock

File: langkit/all_metrics.py [hunk 1]
```diff
@@ -11,14 +13,14 @@
...
-def init() -> DeclarativeSchema:
-    injections.init()
-    topics.init()
-    regexes.init()
-    sentiment.init()
-    textstat.init()
-    themes.init()
-    toxicity.init()
-    input_output.init()
+def init(config: Optional[LangKitConfig] = None) -> DeclarativeSchema:
```
Comment (richard-rogers) [inline langkit/all_metrics.py:16]: Schema factory functions should take all the schema arguments.

File: langkit/count_regexes.py [hunk 2]
```diff
@@ -61,11 +62,12 @@ def _register_udfs():
 def init(
     pattern_file_path: Optional[str] = None, config: Optional[LangKitConfig] = None
 ):
-    config = config or lang_config
+    config = config or deepcopy(lang_config)
```
Comment (richard-rogers) [inline langkit/count_regexes.py:65]: mutating the `config` argument probably isn't great

File: langkit/__init__.py [hunk 3]
```diff
@@ -35,6 +36,12 @@ class LangKitConfig:
         ]
     )
     reference_corpus: str = ""
+    injections_model_path: str = "JasperLS/gelectra-base-injection"
+    rouge_type: str = "rouge1"
+    sentiment_lexicon: str = "vader_lexicon"
+    topic_model_path: str = "MoritzLaurer/mDeBERTa-v3-base-xnli-multilingual-nli-2mil7"
+    topic_classifier: str = "zero-shot-classification"
+    toxicity_model_path: str = "martin-ha/toxic-comment-model"
```
Comment (jamie256) [inline langkit/__init__.py:44]: Not specific to your changes, but we should follow up on what the requirements are for swapping out some of these models, and have some examples.

**Review by jamie256 — APPROVED** [review]
Approach looks fine.

I like the added type hints and incidental cleanup along with making init consistently take in an optional config.

File: langkit/all_metrics.py (hunk 1 above)
Comment (jamie256) [reply langkit/all_metrics.py:16]: Maybe. I think this wrapper of metrics represents something like a preset of metric collection. It is ok if this is a simplification over the degrees of freedom in generating a declarative schema.

Comment (richard-rogers) [reply langkit/all_metrics.py:16]: Then I'd separate the concerns... have the collective `init()`s just initialize the modules and use `udf_schema()` to get the schema.

**Pull request merged**